builtin/providers/google/resource_container_cluster.go

```diff
@@ -142,7 +142,56 @@ func resourceContainerCluster() *schema.Resource {
 				Default:  "default",
 				ForceNew: true,
 			},
-
+			"subnetwork": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"addons_config": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				MaxItems: 1,
```

Comment:
It still doesn't make reading the values from a singleton easier though - one still has to parse it the same way as any other `TypeList`. At least this eliminates the need for `> 1` conditionals, so there's a certain progress. :smiley: